src/main/cpp/checkpoint/CheckPoint.cpp

```diff
@@ -0,0 +1,675 @@
+
+	H5Fclose(newFile);
+	H5Gclose(group);
+}
+
+Population CheckPoint::LoadCheckPoint(boost::gregorian::date date, ClusterStruct& clusters)
+{
+
+	std::string groupname = to_iso_string(date);
+	std::string name = groupname + "/Population";
+	htri_t exist = H5Lexists(m_file, name.c_str(), H5P_DEFAULT);
+	if (exist <= 0) {
+		FATAL_ERROR("Incorrect date loaded");
+	}
+	// loading people
+	hid_t dset = H5Dopen(m_file, name.c_str(), H5P_DEFAULT);
+	hid_t dspace = H5Dget_space(dset);
+	hid_t newType = H5Dget_type(dset);
+
+	hsize_t dims;
+	H5Sget_simple_extent_dims(dspace, &dims, nullptr);
+
+	Population result;
+	for (hsize_t i = 0; i < dims; i++) {
+		hsize_t start = i;
+
+		hsize_t count = 1;
+
+		hid_t subspace = H5Screate_simple(1, &count, nullptr);
+
+		H5Sselect_hyperslab(dspace, H5S_SELECT_SET, &start, nullptr, &count, nullptr);
+
+		personData data;
+
+		H5Dread(dset, newType, subspace, dspace, H5P_DEFAULT, &data);
+
+		disease::Fate disease;
+		disease.start_infectiousness = data.StartInf;
+		disease.start_symptomatic = data.StartSympt;
+		disease.end_infectiousness = data.EndInf;
+		disease.end_symptomatic = data.EndSympt;
+
+		Person toAdd(
+		    data.ID, data.Age, data.Household, data.School, data.Work, data.Primary, data.Secondary, disease);
+
+		if (data.Participating) {
+			toAdd.ParticipateInSurvey();
+		}
+
+		if (data.Immune) {
+			toAdd.GetHealth().SetImmune();
+		}
+		if (data.Infected) {
+			toAdd.GetHealth().StartInfection();
+		}
+		for (unsigned int i = 0; i < data.TimeInfected; i++) {
+			toAdd.GetHealth().Update();
+		}
+
+		result.emplace(toAdd);
+		H5Sclose(subspace);
+	}
+
+	H5Sclose(dspace);
+	H5Tclose(newType);
+	H5Dclose(dset);
+
+	// loading clusters
+	LoadCluster(clusters.m_households, ClusterType::Household, groupname, result);
+	LoadCluster(clusters.m_school_clusters, ClusterType::School, groupname, result);
+	LoadCluster(clusters.m_work_clusters, ClusterType::Work, groupname, result);
+	LoadCluster(clusters.m_primary_community, ClusterType::PrimaryCommunity, groupname, result);
+	LoadCluster(clusters.m_secondary_community, ClusterType::SecondaryCommunity, groupname, result);
+
+	return result;
+}
+
+void CheckPoint::LoadCluster(
+    std::vector<Cluster>& clusters, const ClusterType& i, const std::string& groupname, const Population& result)
+{
+	std::string type = ToString(i);
+	std::string path = groupname + "/" + type;
+	hid_t clusterID = H5Dopen2(m_file, path.c_str(), H5P_DEFAULT);
+	hid_t dspace = H5Dget_space(clusterID);
+	hid_t newType = H5Dget_type(clusterID);
+
+	hsize_t dims;
+	H5Sget_simple_extent_dims(dspace, &dims, nullptr);
+
+	hsize_t start = 0;
+
+	hsize_t count = 1;
+
+	hid_t subspace = H5Dget_space(clusterID);
+	H5Sselect_hyperslab(subspace, H5S_SELECT_SET, &start, nullptr, &count, nullptr);
+	clusterData data;
+
+	H5Dread(clusterID, newType, H5S_ALL, subspace, H5P_DEFAULT, &data);
+
+	H5Sclose(subspace);
+
+	Cluster* CurrentCluster = new Cluster(data.ID, i);
```

Comment:
This is a memory leak. If you absolutely must use a pointer to a `Cluster`, use a `std::unique_ptr<Cluster>` and rewrite this line as `std::unique_ptr<Cluster> current_cluster = std::make_unique<Cluster>(data.ID, i);`